app/src/main/java/com/randomappsinc/studentpicker/common/SpeechToTextManager.java

```diff
@@ -0,0 +1,127 @@
+package com.randomappsinc.studentpicker.common;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.speech.RecognitionListener;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.view.View;
+import android.widget.TextView;
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.joanzapata.iconify.widget.IconTextView;
+import com.randomappsinc.studentpicker.R;
+import java.util.List;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class SpeechToTextManager implements RecognitionListener {
+    public interface Listener {
+        void onSpokenText(String spokenText);
+    }
+
+    @BindView(R.id.voice_icon) IconTextView voiceIcon;
+    @BindView(R.id.message) TextView message;
+    @BindView(R.id.try_again) View tryAgain;
+
+    private SpeechRecognizer speechRecognizer;
```

Comment:
Don't we need to release/destroy this resource somewhere? We can expose a cleanUp() API in this class that the fragment/activity calls in their onDestroy() block. Otherwise, I think we might leak something?